Use the cluster name in the names of the firewall rules that allow cluster-internal traffic to disambiguate the rules belonging to different clusters.

madhusudancs opened the pull request:
Also dropping the network name from these firewall rule names.

Network name was used to disambiguate firewall rules in a given network.
However, since two clusters cannot share a name in a GCE project, this
sufficiently disambiguates the firewall rule names. A potential confusion
arises when someone tries to create a firewall rule with the same name
in a different network, but that's also an indication that they shouldn't
be doing that.


@jszczepkowski due to PR #33094
@ixdy for test-infra

cc @kubernetes/sig-federation @nikhiljindal

madhusudancs committed:
Use the cluster name in the names of the firewall rules that allow cluster-internal traffic to disambiguate the rules belonging to different clusters.

k8s-reviewable commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.kubernetes.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/38638)
<!-- Reviewable:end -->

ixdy commented:
Please update https://github.com/madhusudancs/kubernetes/blob/d92cf4df5edd65f5234e83228c3c1f921ceae968/cluster/gce/util.sh#L1455-L1460 too.

I'm also worried that the firewall rule names might be too long now.

eparis commented:
assigning to ixdy because while it LGTM I was also going to ask about length issues and had no way to test/verify/know the safety.

ixdy commented:
Yeah, it's too long:
```
W1212 11:26:37.617] ERROR: (gcloud.compute.firewall-rules.create) Some requests did not succeed:
W1212 11:26:37.617]  - Invalid value for field 'resource.name': 'e2e-gce-agent-pr-59-0-e2e-gce-agent-pr-59-0-default-internal-master'. Must be a match of regex '(?:[a-z](?:[-a-z0-9]{0,61}[a-z0-9])?)'
W1212 11:26:37.617] 
W1212 11:26:38.118] ERROR: (gcloud.compute.firewall-rules.create) Some requests did not succeed:
W1212 11:26:38.118]  - Invalid value for field 'resource.name': 'e2e-gce-agent-pr-59-0-e2e-gce-agent-pr-59-0-default-internal-node'. Must be a match of regex '(?:[a-z](?:[-a-z0-9]{0,61}[a-z0-9])?)'
```

On Jenkins at least, I think we set `NETWORK` and `CLUSTER_NAME` to the same value, the Jenkins agent we're running tests from.

jszczepkowski commented:
I'm ok with the modification.

madhusudancs committed:
Dropping the network name from the internal master and node firewall rules.

Network name was used to disambiguate firewall rules in a given network.
However, since two clusters cannot share a name in a GCE project, this
sufficiently disambiguates the firewall rule names. A potential confusion
arises when someone tries to create a firewall rule with the same name
in a different network, but that's also an indication that they shouldn't
be doing that.

madhusudancs commented:
@ixdy @eparis the name length limit is somewhat unfortunate. It forces people to trade-off vowels for 140 chrctrs, if you know what I mean ;-)

Anyway, after some thought, I have decided to drop the network name prefix. I have put the reasoning/explanation in the PR description above. Cluster names in a given project should be unique, otherwise bad things will happen. So that should make the firewall rule names unique. The scope of the rule isn't clear in the name now, but I hope the network column in the cloud console helps clarify that.

ixdy commented:
/lgtm

eparis commented:
I removed LGTM, but you can put it back if you so choose. I know. for. a. fact. that people run multiple kubernetes clusters in the same GCE project. I know it can bite them in the bum around storage dynamic provisioning and I hear about it. Do we really need the `default-internal-master` ? Is that enough characters saved to use both?

I don't want to blindly accept that it is ok to break multiple clusters in one project. If this doesn't really break them, so be it. But do not make this change without considering the impact that we know it will have on people...

madhusudancs commented:
@eparis I am not entirely sure about the argument you are making. What does it break? What's the known impact that this will have?

madhusudancs commented:
@eparis btw, I would also like to point out or clarify that this is to exactly enable what you are arguing for, i.e. multiple clusters in a single project, not otherwise. It is broken right now, without this change.

eparis commented:
I don't intend to 'argue' anything. I'm `like a child who wanders into the middle of a movie and wants to know` what's going on (extra points if you get the reference). I read `Cluster names in a given project should be unique` and didn't actually pay attention to the words that were said. I should not have opened my mouth at all. re-adding LGTM

madhusudancs commented:
@eparis no worries. Asking questions is good. It made me think twice and think hard in this case, which is good. Sorry if I came off as hostile. Did not mean that way. I was trying to understand the consequences, particularly the ones I had not seen, if any.
